pkg/feeder/feeder.go

```diff
@@ -52,15 +54,18 @@ func NewClient(baseURL, baseAPI string, client *HttpClient) *Client {
 	}
 
 	// retry mechanism for do requests
+	// https://cloud.google.com/iot/docs/how-tos/exponential-backoff
 	retryFuncForDoReq := func(req *http.Request, httpClient HttpClient) (*http.Response, error) {
 		var res *http.Response
-		wait := 2 * time.Second
+		wait := 1*time.Second + time.Duration(rand.Intn(1000))*time.Millisecond
+		maxWait := float64(10 * time.Minute)
```

Comment:
Do the cast is the function that requires it, not here.

Reply:
Not sure I understand. Do you mean, I can let the `wait` variable be float and cast it to Duration when calling `time.Sleep()`?

Reply:
Try something like this:

```go

	retryFuncForDoReq := func(req *http.Request, httpClient HttpClient) (*http.Response, error) {
		var res *http.Response

		const maxWait = 10 * time.Minute
		wait := 1 * time.Second

		for {
			res, err = httpClient.Do(req)
			if err != nil || res == nil || res.StatusCode != http.StatusOK {
				Logger.With("wait", wait.Seconds(), "error", err).Info("Retrying request")
				time.Sleep(wait)
				wait = time.Duration(math.Min((wait * 2).Seconds(), maxWait.Minutes()))
				continue
			}
			if res.StatusCode == http.StatusOK {
				break
			}
		}
		
		return res, err
	}
```